sqlalchemy_redshift/dialect.py

```diff
@@ -659,13 +659,15 @@ def _get_table_or_view_names(self, relkind, connection, schema=None, **kw):
     def _get_column_info(self, *args, **kwargs):
         kw = kwargs.copy()
         encode = kw.pop('encode', None)
-        if sa_version < Version('1.2.0'):
-            # SQLAlchemy 1.2.0 introduced the 'comment' param
-            del kw['comment']
         if sa_version >= Version('1.3.16'):
             # SQLAlchemy 1.3.16 introduced generated columns,
             # not supported in redshift
             kw['generated'] = ''
+        if sa_version <= Version('1.4.0'):
+            del kw['identity']
+        else:
+            kw['identity'] = None
```

Comment:
@jklukas I'm not the function that finds these `kwargs` is deep in the Redshift/Postgres innards and I don't have familiarity with it right now, so I just didn't support this argument. If you think that's a problem and can point me in the right direction, I can take a look.

Reply:
What's the problem that this is addressing? Is existence of an "identity" kwarg causing failures?

Reply:
Well, you are probably right that I don't need to do this, however, this is called from `get_columns` which eventually uses the following query to get the `kwargs` to this function.

```sql
            SELECT
              n.nspname as "schema",
              c.relname as "table_name",
              att.attname as "name",
              format_encoding(att.attencodingtype::integer) as "encode",
              format_type(att.atttypid, att.atttypmod) as "type",
              att.attisdistkey as "distkey",
              att.attsortkeyord as "sortkey",
              att.attnotnull as "notnull",
              pg_catalog.col_description(att.attrelid, att.attnum)
                as "comment",
              adsrc,
              attnum,
              pg_catalog.format_type(att.atttypid, att.atttypmod),
              pg_catalog.pg_get_expr(ad.adbin, ad.adrelid) AS DEFAULT,
              n.oid as "schema_oid",
              c.oid as "table_oid"
            FROM pg_catalog.pg_class c
            LEFT JOIN pg_catalog.pg_namespace n
              ON n.oid = c.relnamespace
            JOIN pg_catalog.pg_attribute att
              ON att.attrelid = c.oid
            LEFT JOIN pg_catalog.pg_attrdef ad
              ON (att.attrelid, att.attnum) = (ad.adrelid, ad.adnum)
            WHERE n.nspname !~ '^pg_'
              AND att.attnum > 0
              AND NOT att.attisdropped
            UNION
            SELECT
              view_schema as "schema",
              view_name as "table_name",
              col_name as "name",
              null as "encode",
              col_type as "type",
              null as "distkey",
              0 as "sortkey",
              null as "notnull",
              null as "comment",
              null as "adsrc",
              null as "attnum",
              col_type as "format_type",
              null as "default",
              null as "schema_oid",
              null as "table_oid"
            FROM pg_get_late_binding_view_cols() cols(
              view_schema name,
              view_name name,
              col_name name,
              col_type varchar,
              col_num int)
            ORDER BY "schema", "table_name", "attnum";
```

I don't know enough about these tables to know where to get the `identity` information from. I will remove this `else`, but any guidance on getting `identity` information in the query above would be useful. Or if you think I can leave this functionality out, then that's fine too.

Reply:
Does 1.4 require that "identity" exist in kwargs? If so, it's probably fine to set this to `None` if no "identity" key already exists.

Reply:
No, it's not required. I'll make the change.